File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        seen = set()
        for i in range(9):
            for j in range(9):
                if board[i][j] != '.':
                    if (board[i][j] + " row " + str(i)) in seen or \
                       (board[i][j] + " column " + str(j)) in seen or \
                       (board[i][j] + " sub box " + str(i // 3) + str(j // 3)) in seen:
                        return False
                    seen.add(board[i][j] + " row " + str(i))
                    seen.add(board[i][j] + " column " + str(j))
                    seen.add(board[i][j] + " sub box " + str(i // 3) + str(j // 3))
        return True

File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        HashSet<String> h = new HashSet<>();
        for(int i=0; i<9; i++)
        {
            for(int j=0; j<9; j++)
            {
                if(board[i][j]=='.')
                {
                    continue;
                }
                else if(!h.add(board[i][j]+" row "+i) || !h.add(board[i][j]+" column "+j) || 
                !h.add(board[i][j]+" sub box "+i/3+j/3))
                    return false;
            }
        }
        return true;
    }
}